src/Core/src/Graphics/CompareExtensions.cs

```diff
@@ -9,13 +9,14 @@ internal static class CompareExtensions
 			if (max.CompareTo(min) < 0)
 				return max;
 
-			if (value.CompareTo(min) < 0)
-				return min;
+            int compare = value.CompareTo(min);
+            if (compare < 0)
+                return min;
 
-			if (value.CompareTo(max) > 0)
-				return max;
+            if (compare == 0 || value.CompareTo(max) <= 0)
```

Comment:
The old algorithm was perfectly correct and prettier. 

Yet, compare == 0 allows you to avoid the next CompareTo if it is zero, and just return the value.

CompareTo doesn't necessarily mean equals, so I cannot just do <=

It could be for sorting a list of objects that are passed by reference, and thus they would be sorted in the same place, but different objects.

Reply:
If T had a constraint that it was a struct, then I could do it the way you said.